### DISCUSSION TITLE
Multi Tenancy tutorial issue with claims principle

### DISCUSSION THREAD

**Discussion by NadeanB** [serenity-is/Serenity#7078] [Announcements]
### Before submitting the bug report, please read and check the following items

- [X] This isn't a feature request or a discussion topic (Please use discussions for that)
- [X] I have searched open and closed issues for duplicates
- [X] I have searched the documentation on the Serenity Platform website https://serenity.is/docs/
- [X] I have read the frequently asked questions https://serenity.is/docs/startsharp/faq
- [X] I have searched my problem in the discussions and the old issues https://github.com/serenity-is/Serenity/discussions

### What happened?

I am following the Serenity Multi Tenancy tutorial and was asked to post the issue here. There is a problem with the claims principle or the extension method GetTenantId() under the "Filtering users by tenantId" section. It always gives the null error mentioned in the extension method.

### What did you expect to happen?

The extension method should access the tenant id claim successfully.

### How to reproduce?

Follow the Multi Tenancy tutorial.

### What Serenity Nuget Versions are you seeing the problem on? (separated by comma)

6.9.6.4

### Code editor

VS 2022

### Operating System

Windows 11

### On which device do you see the problem?

Laptop

### On which operating system do you see the problem?

Windows

**Comment by VictorTomaili** [comment]
Hello,
In same sample, you can see the static method named `CreatePrincipal`. This function is moved to the `DefaultUserClaimCreator` with interface `IUserClaimCreator`. You can create a new class `MyProjectUserClaimCreator` which extends from `DefaultUserClaimCreator` and override the function with name `CreatePrincipal`. Do not modify existing behavior, just add additional tenant claim. After that do not forget to replace the `DefaultUserClaimCreator` with this new class in your `startup.cs` file. 

Startup.cs File
```diff
- services.AddSingleton<IUserClaimCreator, DefaultUserClaimCreator>();
+ services.AddSingleton<IUserClaimCreator, MyProjectUserClaimCreator>();
```

**Reply by NadeanB** [reply]
Thanks for the reply. Unfortunately the same issue persists after extending `DefaultUserClaimCreator` and overriding `CreatePrinciple`. The error from the `GetTenantId()` in the tutorial still occurs.
`var tenantClaim = user.Claims.FirstOrDefault(x => x.Type == "TenantId");
if (tenantClaim is null)
    throw new NullReferenceException("TenantId claim not found");` 
Do you have any other advice on what else to look for?

**Reply by VictorTomaili** [reply]
Where did you use the GetTenantId function? This function will return null where the HttpContext does not exist like background tasks etc.
